filter/ip2location/filterip2location.go

```
@@ -0,0 +1,197 @@
+var DefaultCIDR = []string{
+	"10.0.0.0/8",
+	"100.64.0.0/10",
+	"127.0.0.0/8",
+	"172.16.0.0/12",
+	"192.168.0.0/16",
+	"fc00::/7",
+	"fe80::/10",
+	"169.254.0.0/16",
+}
+
+// DefaultFilterConfig returns an FilterConfig struct with default values
+func DefaultFilterConfig() FilterConfig {
+	return FilterConfig{
+		FilterConfig: config.FilterConfig{
+			CommonConfig: config.CommonConfig{
+				Type: ModuleName,
+			},
+		},
+		Key:         "ip2location",
+		QuietFail:   false, // backwards compatible
+		SkipPrivate: false,
+		CacheSize:   100000,
+	}
+}
+
+// getCallback implements a callback that switches to the new file
+func (fc *FilterConfig) getCallback() func(string) {
+	return func(fn string) {
+		newDb, err := ip2location.OpenDB(fn)
+		if err != nil {
+			goglog.Logger.Errorf("ip2location failed to update %s: %s", fn, err.Error())
+			return
+		}
+		oldDb := fc.db
+		fc.dbMtx.Lock()
+		fc.db = newDb
+		fc.dbMtx.Unlock()
+		oldDb.Close()
+		fc.cache.Purge()
+		goglog.Logger.Infof("ip2location reloaded file %s", fn)
+	}
+}
+
+// InitHandler initialize the filter plugin
+func InitHandler(ctx context.Context, raw *config.ConfigRaw) (config.TypeFilterConfig, error) {
+	conf := DefaultFilterConfig()
+	err := config.ReflectConfig(raw, &conf)
+	if err != nil {
+		return nil, err
+	}
+
+	// open database
+	conf.db, err = ip2location.OpenDB(conf.DBPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// init cache
+	conf.cache, err = lru.New(conf.CacheSize)
+	if err != nil {
+		return nil, err
+	}
+
+	goglog.Logger.Info("ip2location filewatcher initialized for", conf.DBPath)
+	// init filewatcher
+	_, err = NewFileWatcher(ctx, conf.DBPath, conf.getCallback(), time.Minute)
+	if err != nil {
+		goglog.Logger.Errorf("ip2location failed to init watcher, %s", err.Error())
+	}
+
+	var cidrs []string
+	if len(conf.PrivateNet) > 0 {
+		cidrs = conf.PrivateNet
+	} else {
+		cidrs = DefaultCIDR
+	}
+
+	for _, cidr := range cidrs {
+		_, privateCIDR, _ := net.ParseCIDR(cidr)
```

Should the error be ignored?